core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -587,41 +594,58 @@ private int countProfiles (ArrayList<GeneticProfile> profileList, GeneticAlterat
+	private Boolean hasVirtualStudy = false;
 
-	public CohortDetails(String inputCohortId) {
-		cohortId = inputCohortId;
-		studySampleMap = filterStudySampleMap(getInputStudySampleMap(inputCohortId));
+	public CohortDetails(String[] inputCohortIds) {
+        // cohortId = inputCohortId;
+		studySampleMap = filterStudySampleMap(getInputStudySampleMap(inputCohortIds));
 	}
 
-	private Map<String, Set<String>> getInputStudySampleMap(String inputCohortId) {
+	private Map<String, Set<String>> getInputStudySampleMap(String[] inputCohortIds) {
+
 		Map<String, Set<String>> studySampleMap = new HashMap<>();
-		if(!inputCohortId.toLowerCase().equals("all")){
-			try {
-				SessionServiceUtil sessionServiceUtil = new SessionServiceUtil();
-				CancerStudy cancerStudy = DaoCancerStudy.getCancerStudyByStableId(inputCohortId);
-				if (cancerStudy == null) {
-					Cohort virtualCohort = sessionServiceUtil.getVirtualCohortData(inputCohortId);
-					if (virtualCohort != null && virtualCohort.getCohortStudyCasesMap().size() > 0) {
-						isVirtualStudy = true;
-						for (CohortStudyCasesMap cohortStudyCasesMap : virtualCohort.getCohortStudyCasesMap()) {
-							studySampleMap.put(cohortStudyCasesMap.getStudyID(), cohortStudyCasesMap.getSamples());
-						}
-					} else {
-						System.out.println("virtual study is null");
-					}
-				} else {
-					studySampleMap.put(cancerStudy.getCancerStudyStableId(), new HashSet<String>());
-				}
-			} catch (DaoException e) {
-				e.printStackTrace();
-			}
-		}
+        SessionServiceUtil sessionServiceUtil = new SessionServiceUtil();
+
+        for (String inputCohortId: inputCohortIds) {
+            if(!inputCohortId.toLowerCase().equals("all")){
+                try {
+                    CancerStudy cancerStudy = DaoCancerStudy.getCancerStudyByStableId(inputCohortId);
+                    if (cancerStudy == null) { // is virtual study
+                        Cohort virtualCohort = sessionServiceUtil.getVirtualCohortData(inputCohortId);
+                        if (virtualCohort != null && virtualCohort.getCohortStudyCasesMap().size() > 0) {
+                            hasVirtualStudy = true;
+                            for (CohortStudyCasesMap cohortStudyCasesMap : virtualCohort.getCohortStudyCasesMap()) {
+                                if (studySampleMap.containsKey(cohortStudyCasesMap.getStudyID())) {
+                                    String _sharedStudyId = cohortStudyCasesMap.getStudyID();
+                                    if (studySampleMap.get(_sharedStudyId).size() != 0) { // original entry does not contain all samples
+                                        Set<String> mergedSet = mergeSets(studySampleMap.get(_sharedStudyId), cohortStudyCasesMap.getSamples());
+                                        studySampleMap.put(cohortStudyCasesMap.getStudyID(), mergedSet);                                        
+                                    } // otherwise keep the value as empty set
```

Comment:
What if `studySampleMap.get(_sharedStudyId).size() == 0` 
Consider querying two cohorts(1 regular and 1 vc) cohorts=acc_tcga%2C58b75c50e4b0104c33d0b95d
and consider that vc has acc_tcga study in it.

In this case, I think it final `studySampleMap` should be something like
`{"acc_tcga":<acc_tcga study mapped samples from that vc>, "study2":......}`

I mean there should be an 
```
else {
    studySampleMap.put(cohortStudyCasesMap.getStudyID(), _sharedStudyId);
}
```